### PR TITLE
Fix LayerNorm normalization.

### LINKED ISSUE #2185: LayerNorm does not normalize its input correctly
**Describe the bug**
Both from pytorch documentation...:
https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html#torch.nn.LayerNorm
```
y = (x − E[x]) / sqrt(Var[x] + ϵ)∗γ + β
```

... and from pytorch ATen implementation...:
https://github.com/pytorch/pytorch/blob/6564e746ed494505be5f65aa99f8cf1884bb6342/aten/src/ATen/native/cpu/layer_norm_kernel.cpp#L115
```python
  rstd_val = float(1) / std::sqrt(rstd_val + eps)
```

... they normalize by `sqrt(var + eps)`. Now if we take a look at burn...:
https://github.com/tracel-ai/burn/blob/c29ed43441fc5b3286749816c9069ed90f178a77/crates/burn-core/src/nn/norm/layer.rs#L70

... we normalize by `sqrt(var) + eps`, which leads to precision issues.

### PR DESCRIPTION
**Pull request by booti386**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

Fixes #2185 

### Testing

Add a new test with a large epsilon value.

### CODE REVIEW

**Review by laggui — APPROVED** [review]
Thank you for fixing it so quickly! 🙏
